### ISSUE TITLE
Warn if apps use the view

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#1987]
See discussion in https://github.com/nextcloud/server/pull/1233

Apps should not use private classes. We should start warning in the code checker and maybe even the logs. Apps should use the Node API. And if stuff is missing from there we should invest time in adding that functionality.

The node API was introduced ~3 years ago. So I'd say apps should migrate ASAP!

CC: @icewind1991 @nickvergessen

**Comment by MariusBluem** [comment]
Maybe we should also add this as a policy for apps published over our appstore 🤔 @BernhardPosselt

**Comment by BernhardPosselt** [comment]
@MariusBluem nope, definitely not a goal there. The last thing that we want is to run a PHP parser over abitrary code.

**Comment by nickvergessen** [comment]
No, also the Node API is not always suitable atm.

**Comment by BernhardPosselt** [comment]
Isn't this already integrated into core btw :D?

Maybe the way to go is to just modify the default .travis.yml like this https://github.com/nextcloud/news/blob/master/.travis.yml#L48

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by nickvergessen** [comment]
Yeah it's still heavily used as the node api doesn't allow to do all/the same things the view can
